test/deployframework/framework.go

```diff
@@ -0,0 +1,275 @@
+// needed to customize a ReportingFramework object
+type ReportingFrameworkConfig struct {
+	Namespace                   string
+	KubeConfigPath              string
+	UseKubeProxyForReportingAPI bool
+	UseRouteForReportingAPI     bool
+	HTTPSAPI                    bool
+	ReportingAPIURL             string
+	RouteBearerToken            string
+}
+
+// New is the constructor function that creates and returns a new DeployFramework object
+func New(cfg ReportingFrameworkConfig, logger logrus.FieldLogger, nsPrefix, manifestDir, cleanupScriptPath string) (*DeployFramework, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", cfg.KubeConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to build a kube config from %s: %v", cfg.KubeConfigPath, err)
+	}
+
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to initialize the k8s clientset: %v", err)
+	}
+
+	apiextClient, err := apiextclientv1beta1.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to initialize the apiextensions clientset: %v", err)
+	}
+
+	meteringClient, err := metering.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to initialize the metering clientset: %v", err)
+	}
+
+	deployFramework := &DeployFramework{
+		Logger:            logger,
+		Client:            client,
+		APIExtClient:      apiextClient,
+		MeteringClient:    meteringClient,
+		NamespacePrefix:   nsPrefix,
+		ManifestsDir:      manifestDir,
+		CleanupScriptPath: cleanupScriptPath,
+		Config:            cfg,
+	}
+
+	return deployFramework, nil
+}
+
+// Setup handles the process of deploying metering, and waiting for all necessary resources
+// to become ready in order to proceed with running the reporting tests.
+func (df *DeployFramework) Setup(cfg deploy.Config, targetPods int) (*ReportingFrameworkConfig, error) {
+	var err error
+
+	cfg.OperatorResources, err = deploy.ReadMeteringAnsibleOperatorManifests(df.ManifestsDir, cfg.Platform)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to initialize objects from manifests: %v", err)
+	}
+
+	// randomize namespace and update namespace fields
+	rand.Seed(time.Now().UnixNano())
+	namespace := df.NamespacePrefix + "-" + strconv.Itoa(rand.Intn(50))
+
+	// update the ommitted namespace fields in the test table index
+	df.Config.Namespace = namespace
+	cfg.Namespace = namespace
+	cfg.MeteringConfig.ObjectMeta = meta.ObjectMeta{
+		Name:      "operator-metering",
+		Namespace: namespace,
+	}
+
+	df.Logger.Debugf("Deployer config: %+v", cfg)
+
+	df.Deployer, err = deploy.NewDeployer(cfg, df.Logger, df.Client, df.APIExtClient, df.MeteringClient)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to construct a new deployer object: %v", err)
+	}
+
+	df.Logger.Debugf("Deployer obj: %+v", df.Deployer)
+
+	err = df.Deployer.Install()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to install metering: %v", err)
+	}
+
+	err = df.addE2ENamespaceLabel(namespace)
```

Comment:
Depending on how we want to deal with deleting testing namespaces before running `go test ...`, this is probably not needed.

```bash
# ./hack/e2e.sh
echo "Deleting testing namespaces
kubectl delete ns -l name=e2e-testing
```

Reply:
Yeah, not sure which is best yet.